tensorboard/webapp/widgets/linked_time_fob/linked_time_fob_component.ng.html

```diff
@@ -15,20 +15,16 @@
 limitations under the License.
 -->
 
-<div class="fob" (dblclick)="typeStepRequested()">
-  <span *ngIf="!isTyping"> {{ step | number }} </span>
-  <button
-    aria-label="deselect fob"
-    *ngIf="!isTyping"
-    (click)="removeStep.emit()"
-  >
+<div class="fob">
+  <span
+    contenteditable
```

Comment:
Why contenteditable instead of just an input with type="number"?

A couple reasons I ask:
1. My sense from reading about contenteditable is that it is probably overkill for this job. It's meant for richer text editing. We just want a numeric input. (you wouldn't need to write your own input validator, for example). I get the sense that if you can just use an input then you should just use an input.
2. From an accessibility/markup perspective, input elements are more recommended if you can: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/textbox_role. If you use contenteditable then you have to remember to put roll="textbox".

Reply:
1. In current implementation, we need `<input>` only when it's in "editing mode". To render the html elements(.fob, input, and X button), we need to keep a variable ("isTyping") to decide what elements to render and keeps the variable updated on user actions. With contenteditable, all of these can be removed.

2. To make it look better, we want to have the length of input "fit" nicely in the fob. The length is set by css and we need to update it with js  on key binding and do some pixel calculation; with contenteditable the span element is adjusted automatically and real-time to fit the length of digits

Reply:
Another issue with this being contentEditable is that the cursor shows up when you are dragging. The only solution to this that I see is that we would need to dynamically make it contentEditable which means we would have the isTyping variable or something similar back in.

I do think this looks WAY better than the input box but maybe the best fix there is some css clean up? Just thinking out loud here. This solution does seem cleaner in a lot of ways.